src/main/java/io/pravega/perf/PravegaPerfTest.java

```diff
@@ -44,36 +40,39 @@
  * Data format is in comma separated format as following: {TimeStamp, Sensor Id, Location, TempValue }.
  */
 public class PravegaPerfTest {
-
+    private static final int maxTime = 60 * 60 * 24;
     private static String controllerUri = "tcp://localhost:9090";
     private static int messageSize = 100;
     private static String streamName = StartLocalService.STREAM_NAME;
     private static String scopeName = StartLocalService.SCOPE;
     private static boolean recreate = false;
+    private static boolean writeNread = false;
     private static int producerCount = 0;
     private static int consumerCount = 0;
     private static int segmentCount = 0;
-    private static int events = 3000;
-    private static boolean isRandomKey = false;
+    private static int events = 0;
+    private static int eventsPerSec = 0;
+    private static int eventsPerWorker = 0;
     private static int transactionPerCommit = 0;
     private static int runtimeSec = 0;
     private static final int reportingInterval = 5000;
     private static ScheduledExecutorService bgexecutor;
     private static ForkJoinPool fjexecutor;
-    private static PerfStats produceStats, consumeStats;
+    private static PerfStats produceStats = null;
+    private static PerfStats consumeStats = null;
     private static double throughput = 0;
-    private static ThroughputController tput;
     private static String writeFile = null;
     private static String readFile = null;
+    private static ReaderGroup readerGroup = null;
+    private static final int timeout = 10;
+    private static ClientFactory factory = null;
+    private static ControllerImpl controller = null;
+    private static List<Callable<Void>> readers = null;
```

Comment:
A static mutable list that isn't final with callables (to be executed in threads) is 3 race conditions in 1 line! 
Make this, and all the other variables thread safe. (Preferably by making them instance variables.)

Reply:
I have refactored it to use it as local final variables.